lib.php

```diff
@@ -75,6 +75,23 @@ function tool_cloudmetrics_inplace_editable(string $itemtype, int $itemid, strin
     }
 }
 
+// Returns a calculated default chart period based on the sample interval of $metricname.
+
+function period_from_interval($metricname) {
+    $interval = (int) get_config('tool_excimer', $metricname . '_frequency');
+
+    if ($interval <= 2) {
+        $period = DAYSECS * 7;
+    } else if ($interval <= 120) {
+        $period = DAYSECS * 30;
+    } else if ($interval <= 2880) {
+        $period = DAYSECS * 120;
+    } else {
+        $period = DAYSECS * 365;
+    }
+    return $period;
+}
+
```

Comment:
According to  the Moodle docs.
```
lib.php should not contain any functions other than the API functions (and possibly a few small supporting functions). Any other significant functions should be placed in appropriate classes within [your plugin's classes/ directory].
```
https://docs.moodle.org/dev/lib.php

I would recommend that you create a 'lib' class in classes/metrics, and place this function there.